### PR TITLE
fix: viewport camera grab mouse first time opening a level

### PR DESCRIPTION
**Pull request by djfos**
## What does this PR do?

`ClickDetector::m_detectionState` is not initialized in construction. On Windows, it's fine. But on Linux it would cause the viewport camera grabbing mouse first time opening a level. (m_detectionState is initialized with DetectionState::WaitingForMove)

## How was this PR tested?

Manually tested.

### CODE REVIEW

File: Code/Framework/AzFramework/AzFramework/Viewport/ClickDetector.cpp [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 namespace AzFramework
 {
-    ClickDetector::ClickDetector()
+    ClickDetector::ClickDetector() : m_detectionState(DetectionState::Nil)
```
Comment (mprettner-huawei) [inline Code/Framework/AzFramework/AzFramework/Viewport/ClickDetector.cpp:16]: The [Coding style guide](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#member-initialization) recommends to initialize default values for members in the header file

**Review by mprettner-huawei — CHANGES_REQUESTED** [review]

File: Code/Framework/AzFramework/AzFramework/Viewport/ClickDetector.cpp (hunk 1 above)
Comment (djfos) [reply Code/Framework/AzFramework/AzFramework/Viewport/ClickDetector.cpp]: updated

**Review by mprettner-huawei — APPROVED** [review]

**Comment by mprettner-huawei** [comment]
@djfos Your second commit does not contain the `Signed-off-by` line, and therefore DCO does not pass.

**Review by nick-l-o3de — APPROVED** [review]

**Comment by nick-l-o3de** [comment]
To fix this, you might be able to commit `--amend --signoff` and then force push with `--force-with-lease`

**Pull request merged**